pkg/api/search_tags.go

```diff
@@ -17,6 +22,343 @@ const (
 	ParamScopeIntrinsic = "intrinsic"
 )
 
+// ParseSearchBlockRequest parses all http parameters necessary to perform a block search.
+func ParseSearchBlockRequest(r *http.Request) (*tempopb.SearchBlockRequest, error) {
```

Comment:
can we add tests for these new functions? we have few tests in `pkg/api/search_tags_test.go`. would be nice to have test for these as well.